pyomo/contrib/mindtpy/tests/test_mindtpy_lp_nlp.py

```diff
@@ -0,0 +1,155 @@
+"""Tests for the MINDT solver plugin."""
+from math import fabs
+import pyomo.core.base.symbolic
+import pyutilib.th as unittest
+from pyomo.contrib.mindtpy.tests.eight_process_problem import \
+    EightProcessFlowsheet
+from pyomo.contrib.mindtpy.tests.MINLP_simple import SimpleMINLP as SimpleMINLP
+from pyomo.contrib.mindtpy.tests.MINLP2_simple import SimpleMINLP as SimpleMINLP2
+from pyomo.contrib.mindtpy.tests.MINLP3_simple import SimpleMINLP as SimpleMINLP3
+from pyomo.contrib.mindtpy.tests.from_proposal import ProposalModel
+from pyomo.contrib.mindtpy.tests.online_doc_example import OnlineDocExample
+from pyomo.environ import SolverFactory, value
+
+required_solvers = ('ipopt', 'cplex_persistent')
+required_solvers_temp = ('ipopt', 'cplex')
+if all(SolverFactory(s).available() for s in required_solvers_temp):
+    subsolvers_available = True
+else:
+    subsolvers_available = False
```

Comment:
I believe that `cplex_persistent` is required for these tests.  This is why the tests are failing on Jenkins (on that platform, for some Python versions, we have the lp/nl cplex interface, but not the python bindings, so `cplex` is available, but `cplex_persistent` is not).

This needs to be changed to:
```python
required_solvers = ('ipopt', 'cplex_persistent')
if all(SolverFactory(s).available(False) for s in required_solvers):
    subsolvers_available = True
else:
    subsolvers_available = False
```

Reply:
Thanks for the advice.